src/vs/workbench/contrib/chat/browser/chatQuotasService.ts

```diff
@@ -188,3 +188,50 @@ export class ChatQuotasService extends Disposable implements IChatQuotasService
 		this._onDidChangeQuotas.fire();
 	}
 }
+
+export class ChatQuotasStatusBarEntry extends Disposable implements IWorkbenchContribution {
+
+	static readonly ID = 'chat.quotasStatusBarEntry';
+
+	private readonly _entry = this._register(new MutableDisposable<IStatusbarEntryAccessor>());
+
+	constructor(
+		@IStatusbarService private readonly statusbarService: IStatusbarService,
+		@IChatQuotasService private readonly chatQuotasService: IChatQuotasService
+	) {
+		super();
+
+		this._register(this.chatQuotasService.onDidChangeQuotas(() => this.updateStatusbarEntry()));
+	}
+
+	private updateStatusbarEntry(): void {
+		const { chatQuotaExceeded, completionsQuotaExceeded } = this.chatQuotasService.quotas;
+		if (chatQuotaExceeded || completionsQuotaExceeded) {
+			// Some quota exceeded, show indicator
+			this._entry.value = this.statusbarService.addEntry({
+				name: localize('indicator', "Copilot Quota Indicator"),
+				text: `$(copilot-warning) ${localize('limitReached', "Limit Reached")}`,
+				ariaLabel: localize('copilotQuotaExceeded', "Copilot Limit Reached"),
+				command: OPEN_CHAT_QUOTA_EXCEEDED_DIALOG,
+				kind: 'prominent',
+				showInAllWindows: true,
+				tooltip: quotaToButtonMessage({ chatQuotaExceeded, completionsQuotaExceeded }),
+			}, ChatQuotasStatusBarEntry.ID, StatusbarAlignment.RIGHT, { id: 'GitHub.copilot.status', alignment: StatusbarAlignment.RIGHT });
+		} else {
+			// No quota exceeded, remove indicator
+			if (this._entry.value) {
+				this._entry.clear();
+			}
```

Comment:
Think this is equivalent:

```suggestion
			this._entry.clear();
```